components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/handlers/SessionDataPublisherImpl.java

```diff
@@ -0,0 +1,348 @@
+ */
+
+package org.wso2.carbon.apimgt.keymgt.handlers;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.application.authentication.framework.context.AuthenticationContext;
+import org.wso2.carbon.identity.application.authentication.framework.context.SessionContext;
+import org.wso2.carbon.identity.application.authentication.framework.model.AuthenticatedUser;
+import org.wso2.carbon.identity.core.bean.context.MessageContext;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.carbon.identity.data.publisher.application.authentication.AbstractAuthenticationDataPublisher;
+import org.wso2.carbon.identity.data.publisher.application.authentication.AuthPublisherConstants;
+import org.wso2.carbon.identity.data.publisher.application.authentication.model.AuthenticationData;
+import org.wso2.carbon.identity.data.publisher.application.authentication.model.SessionData;
+import org.wso2.carbon.identity.oauth.IdentityOAuthAdminException;
+import org.wso2.carbon.identity.oauth.OAuthUtil;
+import org.wso2.carbon.identity.oauth.common.exception.InvalidOAuthClientException;
+import org.wso2.carbon.identity.oauth.dao.OAuthAppDAO;
+import org.wso2.carbon.identity.oauth.dao.OAuthAppDO;
+import org.wso2.carbon.identity.oauth.dto.OAuthConsumerAppDTO;
+import org.wso2.carbon.identity.oauth2.IdentityOAuth2Exception;
+import org.wso2.carbon.identity.oauth2.dao.OAuthTokenPersistenceFactory;
+import org.wso2.carbon.identity.oauth2.model.AccessTokenDO;
+import org.wso2.carbon.identity.oauth2.util.OAuth2Util;
+import org.wso2.carbon.user.core.util.UserCoreUtil;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import javax.servlet.http.HttpServletRequest;
+
+import static org.wso2.carbon.identity.oauth.OAuthUtil.handleError;
+
+/**
+ * In this implementation, when a user logout, all the access tokens related to 'admin_store' or 'admin_publisher'
+ * applications for that particular user will be revoked.
+ */
+public class SessionDataPublisherImpl extends AbstractAuthenticationDataPublisher {
+
+    public static final Log log = LogFactory.getLog(SessionDataPublisherImpl.class);
+
+    @Override
+    public void doPublishAuthenticationStepSuccess(AuthenticationData authenticationData) {
+
+    }
+
+    @Override
+    public void doPublishAuthenticationStepFailure(AuthenticationData authenticationData) {
+
+    }
+
+    @Override
+    public void doPublishAuthenticationSuccess(AuthenticationData authenticationData) {
+
+    }
+
+    @Override
+    public void doPublishAuthenticationFailure(AuthenticationData authenticationData) {
+
+    }
+
+    @Override
+    public void doPublishSessionCreation(SessionData sessionData) {
+
+    }
+
+    @Override
+    public void doPublishSessionUpdate(SessionData sessionData) {
+
+    }
+
+    @Override
+    public void doPublishSessionTermination(SessionData sessionData) {
+
```

Comment:
Are these empty methods will be going to implement in the future?

Reply:
Not sure, but since we have to implement all the abstract methods in the implemented abstract class, these empty methods needs be added.